src/main/java/org/owasp/dsomm/metricCA/analyzer/security/filter/TokenFilter.java

```diff
@@ -11,9 +13,33 @@
 @Component
 public class TokenFilter extends OncePerRequestFilter {
 
+  @Autowired
+  private TokenConfiguration tokenConfiguration;
   @Override
   protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) {
 
+    try {
+      final String authorizationHeader = request.getHeader("Authorization");
+      String jwt = null;
+
+      if (authorizationHeader != null && authorizationHeader.startsWith("Bearer ")) {
+        jwt = authorizationHeader.substring(7);
+        if (jwt.equals(tokenConfiguration.getToken())) {
+          System.out.println("Token Present");
+          filterChain.doFilter(request, response);
+        } else {
+          response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+        }
+      } else {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      response.setStatus(401);
+      response.setHeader("Access-Control-Allow-Origin", "*");
```

Comment:
Why do you need a star?

Reply:
Yes, we do not need this line, so I am removing it.